### PR TITLE
[mosaic:gpu] Right-align splat/tiled layout compatibility

### LINKED ISSUE #39998: `splat_is_compatible_with_tiled` left-aligns the shapes
### Description

The docstring for `WGSplatFragLayout.can_broadcast_to` says that "All source dimensions must match the target's trailing dimensions". That is to say, broadcasting is right aligned, just like in numpy. But `splat_is_compatible_with_tiled` doesn't following this: it *left* aligns. Its code is:

```python
def splat_is_compatible_with_tiled(
    l1: fa.WGSplatFragLayout, l2: fa.TiledLayout
) -> bool:
  s1, s2 = l1.shape, l2.base_tile_shape
  return all(d1 % d2 == 0 for d1, d2 in zip(s1, s2))
```

Here's a minimal example of what goes wrong:
```python
from jax.experimental.mosaic.gpu import fragmented_array as fa
from jax.experimental.mosaic.gpu import layouts

# `(32,)` base tile: the minor axis is what it constrains.
reduced = fa.WGMMA_LAYOUT.reduce([0])
splat = fa.WGSplatFragLayout(shape=(4, 8))
print(reduced.base_tile_shape)                                   # (8,)
print(layouts.splat_is_compatible_with_tiled(splat, reduced))    # False; want True
```

### System info (python version, jaxlib version, accelerator, etc.)

```
jax:    0.11.1.dev20260814+ce6cfc5c2b
jaxlib: 0.11.0
numpy:  2.5.1
python: 3.13.5 (main, Jun 12 2025, 12:22:43) [Clang 20.1.4 ]
device info: cpu-1, 1 local devices"
process_count: 1
platform: uname_result(system='Darwin', release='23.5.0', version='Darwin Kernel Version 23.5.0: Wed May  1 20:14:59 PDT 2024; root:xnu-10063.121.3~5/RELEASE_ARM64_T8122', machine='arm64')
```

### PR DESCRIPTION
**Pull request by edluyuan**
## Summary

Fix `splat_is_compatible_with_tiled` to align a splat shape with the trailing dimensions of a tiled layout's base tile.

Before this, the implementation used `zip(s1, s2)`, which left-aligned the two shapes. `TiledLayout` applies its tiling to trailing array dimensions, so this could produce both false negatives and false positives when the splat shape has higher rank than the base tile.

For example, with a base tile shape of `(8,)`:

-  1. `(4, 8)` should be compatible, but was reported as incompatible.
- 2. `(8, 4)` should be incompatible, but was reported as compatible.

The compatibility check is now explicitly offsets into the tiled suffix before checking divisibility.

Extended the regression coverage both cases to the existing strict relayout tests.

Fixes #39998

### CODE REVIEW

**Comment by edluyuan** [comment]
Could a maintainer add the review label when convenient? Thanks!

**Comment by edluyuan** [comment]
Hi! Could a maintainer please add the review label when convenient?

**Review by bchetioui — APPROVED** [review]
Thanks for the fix!

**Comment by edluyuan** [comment]
Hi @bchetioui It looks like there is a lint failure is unrelated to this PR: Pyrefly is failing on two existing `tpu.scan` calls in `jax/_src/pallas/mosaic/sc_primitives.py` because of a missing `dimension` argument. Would you mind rerunning the check when you have a moment?

**Comment by bchetioui** [comment]
> Would you mind rerunning the check when you have a moment?

All good, just needed some handling internally. Thanks again!
